web/components/feed/feed-comments.tsx

```diff
@@ -60,14 +60,14 @@ export function FeedCommentThread(props: {
       parentComment.id && comment.replyToCommentId === parentComment.id
   )
   commentsList.unshift(parentComment)
-  const [inputRef, setInputRef] = useState<HTMLTextAreaElement | null>(null)
+  const inputRef = useRef<HTMLTextAreaElement | null>(null)
   function scrollAndOpenReplyInput(comment: Comment) {
     setReplyToUsername(comment.userUsername)
     setShowReply(true)
-    inputRef?.focus()
+    inputRef.current?.focus()
   }
   useEffect(() => {
-    if (showReply && inputRef) inputRef.focus()
+    if (showReply) inputRef.current?.focus()
   }, [inputRef, showReply])
```

Comment:
`inputRef` not necessary as a dep. Also, the semantics are a little different. This effect might not be re-run like it was before, since the setState would trigger a re-render when the ref was set.

I would just double check that this is working. Probably it is because it waits for `showReply` to be true, and the effect would run after the first render of the input that also assigns the ref.

Reply:
I agree that the semantics changed here. I think your reasoning is correct. I did test it and it seems to be working great. I'll probably triple check before merging that I exercised all the possible code paths.